app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaDetailsPresenter.kt

```diff
@@ -82,25 +82,32 @@ class MangaDetailsPresenter(
     var headerItem = MangaHeaderItem(manga, controller.fromCatalogue)
 
     fun onCreate() {
-
-        isLockedFromSearch = SecureActivityDelegate.shouldBeLocked()
-        headerItem.isLocked = isLockedFromSearch
-        downloadManager.addListener(this)
-        LibraryUpdateService.setListener(this)
-        tracks = db.getTracks(manga).executeAsBlocking()
-        if (manga.source == LocalSource.ID) {
-            refreshAll()
-        } else if (!manga.initialized) {
-            isLoading = true
-            controller.setRefresh(true)
-            controller.updateHeader()
-            refreshAll()
-        } else {
-            updateChapters()
-            controller.updateChapters(this.chapters)
+        scope.launch {
```

Comment:
jay i can remove this if you want but it makes it slightly faster

Reply:
It can stay in scope, but I'll check over time how it performs, my only worry is of a blank page on first load, but this should load quickly